### PR TITLE
Absolute path checks and related changes

### LINKED ISSUE #1376: Home.java:47-48: Prohibit absolute paths in methods...
The puzzle `1352-39528ceb` from #1352 has to be resolved: 

https://github.com/objectionary/eo/blob/09faccc4fba201855269a0b61108053e460c21f3/eo-maven-plugin/src/main/java/org/eolang/maven/Home.java#L47-L48

The puzzle was created by rultor on 31-Oct-22. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by mximp**
closes #1376 

Client code is amended to supply only relative paths into `Home` methods. 
In some cases `Path.relativize` method is used to make relative paths from already built absolute path.
Default `Home` constructor has been removed to make sure base is always provided.

### CODE REVIEW

**Comment by mximp** [comment]
@yegor256 please check and merge

File: eo-maven-plugin/src/main/java/org/eolang/maven/util/Home.java [hunk 1]
```diff
@@ -119,8 +111,10 @@ public void save(final byte[] bytes, final Path path) throws IOException {
      * @param input Input
      * @param path Cwd-relative path to file
      * @throws IOException If fails
+     * @throws IllegalArgumentException If give path is absolute
      */
     public void save(final Input input, final Path path) throws IOException {
+        this.checkRelative(path);
```
[the remark this answers is not in the document]
Comment (mximp) [reply eo-maven-plugin/src/main/java/org/eolang/maven/util/Home.java]: fixed

**Comment by mximp** [comment]
@yegor256 please check again & merge

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**